src/components/ebay-tooltip/component.js

```diff
@@ -1,11 +1,19 @@
 const eventUtils = require('../../common/event-utils');
 
 module.exports = {
-    onInput() {
+    onCreate() {
         this.state = {
             open: false,
         };
     },
```

Comment:
Why are we even handling the state?
It's all based on input.

Reply:
I couldn't get it to work without initializing the state in the `onCreate` method. It just seemed natural to add the state there; at the very least, managing the state the way we previously were in the `onInput` method wasn't working out very well.

Reply:
So then we should have `input` set the state correctly
And then use `state` instead of input

Reply:
Well here's the thing - we do want the state initially set to false. If there are no inputs, the base component emitting events will change the state of `tooltip`. Then in the case that there are inputs, we just want to call `expand` and `collapse` on the base component, which will then emit events that `tooltip` listens to and then changes its own state.

So right now what we are doing is indeed what we want to happen.

Reply:
```
diff --git a/src/components/ebay-tooltip/component.js b/src/components/ebay-tooltip/component.js
index 1e513e38..ed5bb99c 100644
--- a/src/components/ebay-tooltip/component.js
+++ b/src/components/ebay-tooltip/component.js
@@ -7,12 +7,7 @@ module.exports = {
         };
     },
     onInput(input) {
-        if (input.open === true) {
-            this.getComponent('base').expand();
-        }
-        if (input.open === false) {
-            this.getComponent('base').collapse();
-        }
+        this.state.open = input.open;
     },
     handleExpand() {
         this.state.open = true;
```

```
diff --git a/src/components/components/ebay-tooltip-base/component-browser.js b/src/components/components/ebay-tooltip-base/component-browser.js
index 68b986b4..4b4986a9 100644
--- a/src/components/components/ebay-tooltip-base/component-browser.js
+++ b/src/components/components/ebay-tooltip-base/component-browser.js
@@ -23,15 +23,13 @@ module.exports = {
     },

     onInput(input) {
-        if (input.type !== 'tooltip') {
-            if (this._expander) {
-                if (input.open === true) {
-                    this.expand();
-                    this.handleExpand();
-                } else if (input.open === false) {
-                    this.collapse();
-                    this.handleCollapse();
-                }
+        if (this._expander) {
+            if (input.open === true) {
+                this.expand();
+                this.handleExpand();
+            } else if (input.open === false) {
+                this.collapse();
+                this.handleCollapse();
             }
         }
     },
```
This seems to be working locally.

Reply:
I think it is much cleaner and easier to understand that what was before. 
It gets very confusing when there is this `input.type !== "tooltip"`, this component-base should handle all the cases.

I don't see any issues with this, if there is let me know.

Reply:
oh now I see. That makes sense

Reply:
So I tried this and it works mostly... except that the proper events aren't being emitted from the tooltip-base to the tooltip.

Reply:
Are they emitting properly on the other ones or just not tooltip?

Reply:
just not tooltip

Reply:
Then we should take a look why that is, there's something probably missing in tooltip

Reply:
I think the hard thing is that the tooltip is activated on hover, and the hover events aren't changing the state quite correctly.

Reply:
So I pulled in my `8.0.0` event changes to test it. I added `onCollapse` and `onExpand` events to storybook. I see the events are getting fired correctly on tooltip when open toggles and when I hover over. 
I think the only issue I see is it's firing twice? But not a big deal imo.

Reply:
oh okay. What about the escape key press?

Reply:
Works just fine. I would actually change that to do `this.state.open = false` so it rerenders and calls into tooltip base.

Make sure you're focused on the `document`. The trigger is on `document`, which in storybook is in an iframe

Reply:
To test the `open` attribute, what I did is I triggered it to open. Clicked and held on the storybook iframe. This focuses but does not close the tooltip (releasing the mouse button triggers the close). Then I pressed escape and it closed just fine.